lib/ruby-lint/virtual_machine.rb

```diff
@@ -352,6 +352,8 @@ def after_and_asgn
     #
     def on_nth_ref(node)
       var = definitions.lookup(:gvar, "$#{node.children[0]}")
+      # if number not found, then add it as there is no limit for them
+      var = definitions.define_global_variable(node.children[0])
```

Comment:
```rb
var = some_value
var = another_value
```
does not seem right. Did you mean to delete the 1st assignment? Or make the second a `||=`?